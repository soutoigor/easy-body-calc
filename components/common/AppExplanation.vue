<template>
  <article>
    <h1 class="explanation-title">
      <slot name="title" />
    </h1>
    <div class="explanation-content">
      <slot name="content" />
    </div>
  </article>
</template>

<style scoped>
.explanation-title {
  @apply text-5xl lg:text-7xl italic font-black;
}

.explanation-content {
  @apply text-xl flex flex-col gap-3 mt-4;
}
</style>
