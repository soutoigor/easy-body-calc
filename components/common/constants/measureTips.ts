import height from '@/assets/images/height.jpg'
import weight from '@/assets/images/weight.jpg'
import hip from '@/assets/images/hip.jpg'
import neck from '@/assets/images/neck.jpg'
import abdomen from '@/assets/images/abdomen.jpg'

export const measureTips = {
  HEIGHT: {
    TITLE: 'Height',
    IMAGE: height,
    DESCRIPTION: `
      With a measuring tape or a ruler, measure your height in centimeters,
      keep your legs together, your body straight and your head centralized.
    `,
  },
  WEIGHT: {
    TITLE: 'Weight',
    IMAGE: weight,
    DESCRIPTION: `
      Check your weight with a scale, preferably in fasting and with underwear.
    `,
  },
  HIP: {
    TITLE: 'Hip',
    IMAGE: hip,
    DESCRIPTION: `
      With a measuring tape, contour the biggest part of your hip and get the centimeters.
    `,
  },
  NECK: {
    TITLE: 'Neck',
    IMAGE: neck,
    DESCRIPTION: `
      With a measuring tape, contour your neck like a necktie, above the trapezius,
      but careful to not take the trapezius in consideration.
    `,
  },
  ABDOMEN: {
    TITLE: 'Abdomen',
    IMAGE: abdomen,
    DESCRIPTION: `
      Exhale all the air, but don't force the abdomen. And with a measuring tape, contour
      your waist right above your belly.
    `,
  },
}
