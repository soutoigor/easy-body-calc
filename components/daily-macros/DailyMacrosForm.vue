<template>
  <form class="bodyfat-form" @submit.prevent="sendResult">
    <InputNumber v-model="form.calories" label="Daily calories" />
    <span v-if="!form.calories">
      Don't know your <b>Daily Calories</b>?
      <NuxtLink :to="dailyCaloriesLink" class="calories-link">
        Calculate here
      </NuxtLink>
    </span>
    <InputNumber v-model="form.weight" label="Weight (kg)" />
    <InputSelect
      v-model="form.goal"
      label="Select your goal"
      :items="goalItems"
    />
    <AppButton type="submit" data-testid="submit" :disabled="disabled">
      Calculate
    </AppButton>
  </form>
</template>

<script lang="ts">
import InputNumber from '@/components/common/InputNumber.vue'
import InputSelect, { Items } from '@/components/common/InputSelect.vue'
import AppButton from '@/components/common/AppButton.vue'
import { siteMap } from '@/utils/siteMap'
import {
  computed,
  defineComponent,
  reactive,
  onMounted,
  useRoute,
} from '@nuxtjs/composition-api'
import { DailyMacros, Result, Goal } from './interface'

export default defineComponent({
  components: {
    InputNumber,
    AppButton,
    InputSelect,
  },
  emits: ['result'],
  setup(_, { emit }) {
    const route = useRoute()

    const goalItems: Items = [
      {
        label: 'Lose weight',
        value: 'lose',
      },
      {
        label: 'Keep weight',
        value: 'keep',
      },
      {
        label: 'Gain weight',
        value: 'gain',
      },
    ]

    const form = reactive<DailyMacros>({
      calories: 0,
      weight: 0,
      goal: 'lose',
    })

    const disabled = computed<boolean>(() =>
      Object.values(form).some((value) => !value)
    )

    onMounted(() => {
      const { calories } = route.value.query
      if (calories) {
        form.calories = +calories
      }
    })

    const getNewCalories = (goal: Goal, calories: number) => {
      const caloriesByGoal = {
        lose: calories - (calories * 10) / 100,
        keep: calories,
        gain: calories + (calories * 10) / 100,
      }

      return caloriesByGoal[goal]
    }

    const calculateDailyMacros = (form: DailyMacros): Result => {
      const newCalories = getNewCalories(form.goal, form.calories)

      const protein = Math.floor(form.weight * 2.4)
      const fat = Math.floor((newCalories * 25) / 100 / 9)
      const carb = Math.floor((newCalories * 50) / 100 / 4)
      const calories = protein * 4 + fat * 9 + carb * 4

      return {
        protein,
        fat,
        carb,
        calories,
        goal: form.goal,
      }
    }

    const sendResult = () => {
      const result = calculateDailyMacros(form)

      emit('result', result)
    }

    return {
      sendResult,
      form,
      disabled,
      goalItems,
      dailyCaloriesLink: siteMap.CALORIES.url,
    }
  },
})
</script>

<style scoped>
.bodyfat-form {
  @apply flex flex-col gap-8;
}

.calories-link {
  @apply text-turbo font-semibold text-lg underline;
}
</style>
