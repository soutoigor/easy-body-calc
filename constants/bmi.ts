export const BMI_RESULTS = {
  UNDERWEIGHT: 'underweight',
  NORMAL: 'normal',
  OVERWEIGHT: 'overweight',
  OBESE: 'obese',
  EXTREMELY_OBESE: 'extremely obese',
}
