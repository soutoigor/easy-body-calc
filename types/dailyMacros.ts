export type TGoal = 'lose' | 'keep' | 'gain'

export interface IGoals {
  LOSE: TGoal
  KEEP: TGoal
  GAIN: TGoal
}

export interface IDailyMacros {
  calories: number
  weight: number
  goal: TGoal
}

export interface IDailyMacrosResult {
  goal: TGoal
  fat: number
  protein: number
  carb: number
  calories: number
}

export type TResultForm = IDailyMacrosResult | null
