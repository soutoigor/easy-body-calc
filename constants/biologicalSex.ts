import { IBiologicalSex } from '@/types/biologicalSex'
import { TSelectItems } from '@/types/selectOption'

export const BIOLOGICAL_SEX: IBiologicalSex = {
  MALE: 'M',
  FEMALE: 'F',
}

export const BIOLOGICAL_SEX_ITEMS: TSelectItems = [
  {
    label: 'Male',
    value: BIOLOGICAL_SEX.MALE,
  },
  {
    label: 'Female',
    value: BIOLOGICAL_SEX.FEMALE,
  },
]
