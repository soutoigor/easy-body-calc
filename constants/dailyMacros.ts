import { IGoals } from '@/types/dailyMacros'

export const GOALS: IGoals = {
  LOSE: 'lose',
  KEEP: 'keep',
  GAIN: 'gain',
}
