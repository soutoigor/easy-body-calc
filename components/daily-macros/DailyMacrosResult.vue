<template>
  <div>
    <div class="result">
      <div class="result__item">
        <span class="result__value">{{ result.carb }}</span>
        <span class="result__text">Carbs</span>
      </div>
      <div class="result__item">
        <span class="result__value">{{ result.fat }}</span>
        <span class="result__text">Fat</span>
      </div>
      <div class="result__item">
        <span class="result__value">{{ result.protein }}</span>
        <span class="result__text">Protein</span>
      </div>
      <div class="result__item">
        <span class="result__value">{{ result.calories }}</span>
        <span class="result__text">Calories</span>
      </div>
    </div>
    <p class="result__advice">{{ resultAdvice }}</p>
  </div>
</template>

<script lang="ts">
import { defineComponent, PropType, computed } from '@nuxtjs/composition-api'
import { IDailyMacrosResult } from '@/types/dailyMacros'
import { GOALS } from '@/constants/dailyMacros'

const { LOSE, KEEP, GAIN } = GOALS

export default defineComponent({
  props: {
    result: {
      type: Object as PropType<IDailyMacrosResult>,
      required: true,
    },
  },
  setup({ result }) {
    const messageByGoal = {
      [LOSE]:
        'lose weight, at the end of a every week, you have not lose weight, reduce the Carb value in 10% until you reach your goal',
      [KEEP]: 'keep your weight',
      [GAIN]:
        'gain weight, at the end of a every week, you have not, increase the Carb value in 10% until you reach your goal',
    }

    const resultAdvice = computed(
      () => `Reach the macros above everyday to ${messageByGoal[result.goal]}.`
    )

    return {
      resultAdvice,
    }
  },
})
</script>

<style scoped>
.result {
  @apply border border-turbo rounded-lg h-60 w-full;
  @apply flex items-center justify-evenly;
}

.result__item {
  @apply flex flex-col items-center;
}

.result__value {
  @apply text-3xl lg:text-4xl italic font-black text-turbo;
}

.result__text {
  @apply italic capitalize font-bold text-xl lg:text-2xl;
}

.result__advice {
  @apply mt-4 text-xl;
}
</style>
