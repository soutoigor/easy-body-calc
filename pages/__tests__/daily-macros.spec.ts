import { mount, RouterLinkStub } from '@vue/test-utils'
import Vue from 'vue'
import DailyMacros from '@/pages/daily-macros.vue'
import DailyMacrosForm from '@/components/daily-macros/DailyMacrosForm.vue'
import DailyMacrosResult from '@/components/daily-macros/DailyMacrosResult.vue'
import AppExplanation from '@/components/common/AppExplanation.vue'
import { IDailyMacrosResult } from '@/types/dailyMacros'
import { GOALS } from './../../constants/dailyMacros'

const WrapperComponent = Vue.component('WrapperComponent', {
  template: '<div><slot /></div>',
})

jest.mock('@/router', () => ({
  useRoute: jest.fn(() => ({
    value: {
      query: {
        calories: '2000',
      },
    },
  })),
}))

const result: IDailyMacrosResult = {
  goal: GOALS.GAIN,
  fat: 50,
  protein: 120,
  carb: 200,
  calories: 2000,
}

const mountComponent = () =>
  mount(WrapperComponent, {
    stubs: {
      NuxtLink: RouterLinkStub,
    },
    slots: {
      default: DailyMacros,
    },
  })

describe('DailyMacros', () => {
  it('should mount the component', () => {
    const wrapper = mountComponent()

    expect(wrapper.vm).toBeDefined()
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should show form and hide the result when there is no "result"', () => {
    const wrapper = mountComponent()

    expect(wrapper.findComponent(AppExplanation).isVisible()).toBe(true)
    expect(wrapper.findComponent(DailyMacrosForm).isVisible()).toBe(true)
    expect(wrapper.findComponent(DailyMacrosResult).exists()).toBe(false)
  })

  it('should show result and hide form when there is "result"', async () => {
    const wrapper = mountComponent()

    wrapper.findComponent(DailyMacrosForm).vm.$emit('result', result)

    await wrapper.vm.$nextTick()

    expect(wrapper.findComponent(DailyMacrosResult).isVisible()).toBe(true)
    expect(wrapper.findComponent(AppExplanation).exists()).toBe(true)
    expect(wrapper.findComponent(DailyMacrosForm).exists()).toBe(false)
  })

  it('should show "formTitle" according if there is "result"', async () => {
    const wrapper = mountComponent()

    const title = wrapper.find('.calc-section__title')

    expect(title.text()).toContain('Calculate your Macros bellow:')

    wrapper.findComponent(DailyMacrosForm).vm.$emit('result', result)

    await wrapper.vm.$nextTick()

    expect(title.text()).toContain('Your result:')
  })

  it('should display form again if "calculate again" button is clicked', async () => {
    const wrapper = mountComponent()

    wrapper.findComponent(DailyMacrosForm).vm.$emit('result', result)

    await wrapper.vm.$nextTick()

    await wrapper.find('[data-testid="calculate-again"]').trigger('click')

    await wrapper.vm.$nextTick()

    expect(wrapper.findComponent(DailyMacrosForm).isVisible()).toBe(true)
    expect(wrapper.findComponent(AppExplanation).isVisible()).toBe(true)
    expect(wrapper.findComponent(DailyMacrosResult).exists()).toBe(false)
  })

  it('should display macros modal if click "Know more about each macro"', async () => {
    const wrapper = mountComponent()

    const modalTitles = wrapper.find('.macros__title')

    expect(modalTitles.isVisible()).toBe(false)

    await wrapper.find('[data-testid="open-macros"]').trigger('click')

    await wrapper.vm.$nextTick()

    expect(modalTitles.isVisible()).toBe(true)
  })
})
