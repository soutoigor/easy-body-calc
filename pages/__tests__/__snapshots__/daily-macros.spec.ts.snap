// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DailyMacros should mount the component 1`] = `
"<div>
  <section class=\\"calc-section\\">
    <h2 class=\\"calc-section__title\\">Calculate your Macros bellow:</h2>
    <transition-stub name=\\"rotate\\">
      <div class=\\"calc-section__form-container\\">
        <form class=\\"bodyfat-form\\">
          <div><label for=\\"input\\" class=\\"field__label\\">Daily calories</label> <input id=\\"input\\" type=\\"text\\" autocomplete=\\"off\\" class=\\"field__input\\"></div> <span>
    Don't know your <b>Daily Calories</b>?
    <a class=\\"calories-link\\">
      Calculate here
    </a></span>
          <div><label for=\\"input\\" class=\\"field__label\\">Weight (kg)</label> <input id=\\"input\\" type=\\"text\\" autocomplete=\\"off\\" class=\\"field__input\\"></div>
          <div><label for=\\"select\\" class=\\"field__label\\">Select your goal</label> <select id=\\"select\\" data-testid=\\"field\\" class=\\"field__input field__input--active\\">
              <option value=\\"lose\\">
                Lose weight
              </option>
              <option value=\\"keep\\">
                Keep weight
              </option>
              <option value=\\"gain\\">
                Gain weight
              </option>
            </select></div> <button type=\\"submit\\" disabled=\\"disabled\\" class=\\"button primary\\" data-testid=\\"submit\\">
            Calculate
          </button>
        </form>
        <div><button type=\\"button\\" class=\\"button primary text\\" data-testid=\\"open-modal\\">
            How to get the measures
          </button>
          <transition-stub name=\\"modal\\" style=\\"display: none;\\">
            <div data-testid=\\"backdrop\\" class=\\"modal__backdrop\\">
              <article role=\\"dialog\\" aria-labelledby=\\"modalTitle\\" aria-describedby=\\"modalDescription\\" class=\\"modal__container\\">
                <header class=\\"modal__header\\">
                  <h1 class=\\"modal__title\\">How to get the measures</h1> <button data-testid=\\"closeButton\\" class=\\"modal__close\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 20 20\\" fill=\\"currentColor\\" class=\\"modal__close-icon\\">
                      <path fill-rule=\\"evenodd\\" d=\\"M4.293 4.293a1 1 0 011.414 0L10 8.586l4.293-4.293a1 1 0 111.414 1.414L11.414 10l4.293 4.293a1 1 0 01-1.414 1.414L10 11.414l-4.293 4.293a1 1 0 01-1.414-1.414L8.586 10 4.293 5.707a1 1 0 010-1.414z\\" clip-rule=\\"evenodd\\"></path>
                    </svg></button>
                </header>
                <main class=\\"modal__content\\">
                  <div data-testid=\\"collapse\\">
                    <section class=\\"collapse\\">
                      <header class=\\"collapse__header\\">
                        <h1>Height</h1> <svg xmlns=\\"http://www.w3.org/2000/svg\\" fill=\\"none\\" viewBox=\\"0 0 24 24\\" stroke=\\"currentColor\\" class=\\"collapse__icon\\">
                          <path stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" stroke-width=\\"2\\" d=\\"M19 9l-7 7-7-7\\"></path>
                        </svg>
                      </header>
                      <transition-stub name=\\"expand\\">
                        <!---->
                      </transition-stub>
                    </section>
                    <section class=\\"collapse\\">
                      <header class=\\"collapse__header\\">
                        <h1>Weight</h1> <svg xmlns=\\"http://www.w3.org/2000/svg\\" fill=\\"none\\" viewBox=\\"0 0 24 24\\" stroke=\\"currentColor\\" class=\\"collapse__icon\\">
                          <path stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" stroke-width=\\"2\\" d=\\"M19 9l-7 7-7-7\\"></path>
                        </svg>
                      </header>
                      <transition-stub name=\\"expand\\">
                        <!---->
                      </transition-stub>
                    </section>
                    <section class=\\"collapse\\">
                      <header class=\\"collapse__header\\">
                        <h1>Hip</h1> <svg xmlns=\\"http://www.w3.org/2000/svg\\" fill=\\"none\\" viewBox=\\"0 0 24 24\\" stroke=\\"currentColor\\" class=\\"collapse__icon\\">
                          <path stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" stroke-width=\\"2\\" d=\\"M19 9l-7 7-7-7\\"></path>
                        </svg>
                      </header>
                      <transition-stub name=\\"expand\\">
                        <!---->
                      </transition-stub>
                    </section>
                    <section class=\\"collapse\\">
                      <header class=\\"collapse__header\\">
                        <h1>Neck</h1> <svg xmlns=\\"http://www.w3.org/2000/svg\\" fill=\\"none\\" viewBox=\\"0 0 24 24\\" stroke=\\"currentColor\\" class=\\"collapse__icon\\">
                          <path stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" stroke-width=\\"2\\" d=\\"M19 9l-7 7-7-7\\"></path>
                        </svg>
                      </header>
                      <transition-stub name=\\"expand\\">
                        <!---->
                      </transition-stub>
                    </section>
                    <section class=\\"collapse\\">
                      <header class=\\"collapse__header\\">
                        <h1>Abdomen</h1> <svg xmlns=\\"http://www.w3.org/2000/svg\\" fill=\\"none\\" viewBox=\\"0 0 24 24\\" stroke=\\"currentColor\\" class=\\"collapse__icon\\">
                          <path stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" stroke-width=\\"2\\" d=\\"M19 9l-7 7-7-7\\"></path>
                        </svg>
                      </header>
                      <transition-stub name=\\"expand\\">
                        <!---->
                      </transition-stub>
                    </section>
                    <section class=\\"collapse\\">
                      <header class=\\"collapse__header\\">
                        <h1>Daily activities</h1> <svg xmlns=\\"http://www.w3.org/2000/svg\\" fill=\\"none\\" viewBox=\\"0 0 24 24\\" stroke=\\"currentColor\\" class=\\"collapse__icon\\">
                          <path stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" stroke-width=\\"2\\" d=\\"M19 9l-7 7-7-7\\"></path>
                        </svg>
                      </header>
                      <transition-stub name=\\"expand\\">
                        <!---->
                      </transition-stub>
                    </section>
                  </div>
                </main>
              </article>
            </div>
          </transition-stub>
        </div>
      </div>
    </transition-stub>
  </section>
  <aside class=\\"explanation-section\\">
    <article>
      <h1 class=\\"explanation-title\\">DAILY MACROS</h1>
      <div class=\\"explanation-content\\">
        <p>
          There are 3 macronutrients present in our food, they are
          Carbohydrate, Fat and Protein. Each one has its goals to be consumed
          and its characteristics (calories per gram, time to absorb, etc).
        </p>
        <p>
          It's the total of calories that we ingest in the day that makes us
          gain, keep, or lose weight. It's pretty simple, if you ingest more
          than your body needs in a day (Daily calories), you will gain
          weight, and the opposite is true, if you ingest less than you need,
          you lose weight.
        </p>
        <p>
          But our body is smart, when you change your daily calorie intake, it
          adapts to survive with that new total of calories, and stop gaining
          or losing weight, so you need to adapt your macros. Generally, we
          change the Carbohydrate in 10%, up if gaining weight, down if
          losing. you can change the Fat too, if you already changed the
          Carbohydrate too much. If the carbohydrate is too low, add more
          cardio exercises.
        </p>
        <p>
          But you need to do it slowly, unhurried, if you make these changes
          too fast, there will come a time when you will no longer be able to
          add or remove calories in a healthy way, and you will no longer
          evolve.
        </p>
        <p>And you can eat anything you want, if it fits your macros.</p>
      </div>
    </article> <button type=\\"button\\" class=\\"button primary text\\" data-testid=\\"open-macros\\">
      Know more about each macro
    </button>
    <transition-stub name=\\"modal\\" style=\\"display: none;\\">
      <div data-testid=\\"backdrop\\" class=\\"modal__backdrop\\">
        <article role=\\"dialog\\" aria-labelledby=\\"modalTitle\\" aria-describedby=\\"modalDescription\\" class=\\"modal__container\\">
          <header class=\\"modal__header\\">
            <h1 class=\\"modal__title\\">Macronutrients</h1> <button data-testid=\\"closeButton\\" class=\\"modal__close\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 20 20\\" fill=\\"currentColor\\" class=\\"modal__close-icon\\">
                <path fill-rule=\\"evenodd\\" d=\\"M4.293 4.293a1 1 0 011.414 0L10 8.586l4.293-4.293a1 1 0 111.414 1.414L11.414 10l4.293 4.293a1 1 0 01-1.414 1.414L10 11.414l-4.293 4.293a1 1 0 01-1.414-1.414L8.586 10 4.293 5.707a1 1 0 010-1.414z\\" clip-rule=\\"evenodd\\"></path>
              </svg></button>
          </header>
          <main class=\\"modal__content\\">
            <h3 class=\\"macros__title\\">Protein</h3>
            <p>
              Each gram of protein has 4 calories. The function of protein is to
              create and regenerate our body tissues such as skin, nails, hair,
              organs and muscles. Like carbohydrate, it changes the absorption
              time. depending on its origin (example of slow absorption: eggs,
              casein. Example of quick: whey, meat. The ideal is to ingest protein
              from slow absorption before a long period without eating. But that's
              not will cause big differences, worry more about the biological
              value of this protein, proteins from animal sources such as eggs and
              meat have a better biological value than q from whole sources. the
              better the biological value, the better your muscles will absorb.
            </p>
            <h3 class=\\"macros__title\\">Fat</h3>
            <p>
              Each gram of fat has 9 calories. The function of fat is provide
              energy for the body, synthesize hormones, are the vehicle for
              fat-soluble vitamins, compose cellular structures and increase the
              palatability of foods. Your fonts can be both animal and vegetable.
              The main sources of fat are: cheeses, meat, milk, olives,
              chocolates, butter, eggs, etc.
            </p>
            <h3 class=\\"macros__title\\">Carbohydrate</h3>
            <p>
              Carbohydrate has 4 calories per gram. Its function is to provide
              energy and glycogen for muscles, brain, bone marrow and various
              functions of the nervous system. Depending on the source of
              carbohydrate, this changes your glycemic index, which is the speed
              that it is processed and peak amount of insulin it produces.
              Carbohydrate when it is quickly processed and is not used as a
              source of energy, it converts into adipose tissue (its fats), by
              this, avoid foods with a high glycemic index such as: sweets, soft
              drinks, food made with flour such as Paes and pasta. use wisely, put
              this type of food for a post workout, or upon waking up, which is
              when your body really needs to use itself.
            </p>
          </main>
        </article>
      </div>
    </transition-stub>
  </aside>
</div>"
`;
